Python/pylifecycle.c

```diff
@@ -3350,6 +3353,31 @@ wait_for_thread_shutdown(PyThreadState *tstate)
     Py_DECREF(threading);
 }
 
+/* gh-123940: Mark remaining daemon threads as exited so that they may
+ * be joined from finalizers.
+ */
+static void
+daemon_threads_exited(PyThreadState *tstate)
+{
+    PyObject *threading = PyImport_GetModule(&_Py_ID(threading));
```

Comment:
This implementation makes me worry because it's touch the import system during shutdown and adding a new call back into Python after we have force-killed (or deadlocked) daemon threads.

I think @ericsnowcurrently suggested moving `ThreadHandle` to `Python/thread.c`. Would it be possible to refactor this -- maybe with an API exposed in `pycore_pythread.h` -- to avoid the `PyImport_GetModule` and the calls into Python?

Reply:
> Would it be possible to refactor this -- maybe with an API exposed in pycore_pythread.h -- to avoid the PyImport_GetModule and the calls into Python?

Yep, that's possible.